lib/src/components/NxDropdown/stateful/__tests__/NxStatefulDropdown.test.tsx

```diff
@@ -5,192 +5,311 @@
+
+    expect(dropdown).toHaveClass('extra-class');
+    expect(toggleButton).toHaveClass('nx-btn--tertiary');
+    expect(toggleButton).toHaveClass('disabled');
+    expect(toggleButton).toHaveAttribute('aria-disabled');
   });
 
-  it('opens the dropdown when the toggle is clicked', function() {
-    const mounted = getMountedComponent(undefined, { attachTo: container });
+  it('toggles the menu open and close when toggle is clicked', async function() {
+    const user = userEvent.setup();
 
-    expect(mounted.find('.nx-dropdown-menu')).not.toExist();
+    const { container } = quickRender();
+    const toggleButton = screen.getByRole('button', { name: 'dropdown-toggle'});
 
-    mounted.find(NxButton).simulate('click');
-    expect(mounted.find('.nx-dropdown-menu')).toExist();
-    mounted.unmount();
+    let menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
+
+    await user.click(toggleButton);
+
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await user.click(toggleButton);
+
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
   });
 
-  it('closes the dropdown if the menu is open and the toggle is clicked', function() {
-    const mounted = getMountedComponent(undefined, { attachTo: container });
+  it('closes the dropdown when the Escape key is pressed on this component', async function() {
+    const user = userEvent.setup();
+    const { container } = quickRender();
 
-    act(() => {
-      mounted.find(NxButton).getDOMNode().dispatchEvent(new MouseEvent('click', { bubbles: true }));
-    });
-    mounted.update();
-    expect(mounted.find('.nx-dropdown-menu')).toExist();
+    const toggleButton = screen.getByRole('button', { name: 'dropdown-toggle'});
+    let menu = container.querySelector('.nx-dropdown-menu');
 
-    act(() => {
-      mounted.find(NxButton).getDOMNode().dispatchEvent(new MouseEvent('click', { bubbles: true }));
-    });
-    mounted.update();
-    expect(mounted.find('.nx-dropdown-menu')).not.toExist();
-    mounted.unmount();
+    await user.click(toggleButton);
+
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await fireEvent.keyDown(menu as HTMLElement, { key: 'Escape' });
+
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
   });
 
-  it('closes the dropdown when the Escape key is pressed on this component', function() {
-    const mounted = getMountedComponent(undefined, { attachTo: container });
+  it('closes the dropdown when an outside click happens', async function() {
+    const user = userEvent.setup();
 
-    act(() => {
-      mounted.find(NxButton).getDOMNode().dispatchEvent(new MouseEvent('click', { bubbles: true }));
-    });
-    mounted.update();
-    expect(mounted.find('.nx-dropdown-menu')).toExist();
+    const { container } = render(
```

Comment:
This is fine but another option would be to still use `quickRender` or `renderEl` for the dropdown but then also have a separate `render()` statement for the button

Reply:
> This is fine but another option would be to still use quickRender or renderEl for the dropdown but then also have a separate render() statement for the button

Thanks didn't know you could do that: a1fb4380312a886fdb4f7b946d7b797fbd73df25